kubernetes/deployment-test.tmpl

```diff
@@ -0,0 +1,57 @@
+apiVersion: apps/v1
+kind: Deployment
```

Comment:
we don't want this to run in our k8s cluster, as it'll leave the pods running there on a test service end point taking up resources but doing nothing. 

I don't see what value this currently provides via deployment into the cluster, considering we already have a test in deploy gh action workflow that will cause early termination if it fails and guards against deploying borked builds. 

Unless of course this morphs into a staging system deploy instead - that could be useful! 

It could run 1 pod on a public URL say `https://static-staging.zooniverse.org/` and allows us to test our nginx configs before shipping to prod...that could be useful an would shift our deploy paradigm in this repo to align with other services, i.e. 
1. merges to main/master auto deploy to staging
2. intentional deploys via `lita deploy static` that deploy via the `production-release` tag and associated GH action.

Reply:
My intention was not to spin them up and leave them running forever, but to allow me to ensure that this action can build and deploy exactly what is expected in parallel with the production service. Actions always require a bit of tweaking post-merge since there's no good way to test them that doesn't affect the relevant services, so this allowed for that while not affecting our most important production service. I expected to remove the test service/template after that. 

However, I actually love the idea of turning this into a staging service instead. I'll think about how a staging http-frontend service could be set up to proxy all the same stuff and enable us to test more elaborate nginx tweaks before committing to them in production and update this PR.